src/ShopBundle/objects/db/TShopArticle.class.php

```diff
@@ -245,14 +245,16 @@ private function getOwnVat()
     /**
      * return the link to the detail view of the product.
      *
-     * @param bool                $bAbsolute           set to true to include the domain in the link
+     * @param bool                $bAbsolute set to true to include the domain in the link
      * @param null|string         $sAnchor
      * @param array               $aOptionalParameters supported optional parameters:
      *                                                 TdbShopArticle::CMS_LINKABLE_OBJECT_PARAM_CATEGORY - (string) force the article link to be within the given category id (only works if the category is assigned to the article)
      * @param TdbCmsPortal|null   $portal
      * @param TdbCmsLanguage|null $language
      *
      * @return string
+     *
+     * @throws ErrorException
```

Comment:
Unfortunately we can't do this - the interface method doesn't declare the exception and implementations can't throw exceptions that the interface doesn't allow.

We should either avoid the exception altogether, or adjust the interface method in a minor version (but declare a better exception type), but both ways are out of scope for this issue.

Reply:
An exception can be thrown that is not declared, yes. We need to deal with this situation, but in a fix release we could do nothing but catching the exception (if the method wasn't an override of a parent method or the implementation of an interface, I'd say we just take the freedom to add the `throws` declaration).